src/legacy/core_plugins/kibana/public/discover/embeddable/search_embeddable.ts

```diff
@@ -251,31 +255,76 @@ export class SearchEmbeddable extends Embeddable<SearchInput, SearchOutput>
         },
       });
     };
-
-    this.searchScope = searchScope;
   }
 
   public reload() {
-    this.courier.fetch();
+    this.fetch();
   }
 
+  private fetch = async () => {
+    if (!this.searchScope) return;
+
+    const { searchSource } = this.savedSearch;
+    searchSource.setField('size', config.get('discover:sampleSize'));
+    searchSource.setField('sort', getSort(this.searchScope.sort, this.searchScope.indexPattern));
+
+    // Log request to inspector
+    this.inspectorAdaptors.requests.reset();
+    const title = i18n.translate('kbn.embeddable.inspectorRequestDataTitle', {
+      defaultMessage: 'Data',
+    });
+    const description = i18n.translate('kbn.embeddable.inspectorRequestDescription', {
+      defaultMessage: 'This request queries Elasticsearch to fetch the data for the search.',
+    });
+    const inspectorRequest = this.inspectorAdaptors.requests.start(title, { description });
+    inspectorRequest.stats(getRequestInspectorStats(searchSource));
+    searchSource.getSearchRequestBody().then((body: any) => {
+      inspectorRequest.json(body);
+    });
+
+    this.searchScope.isLoading = true;
+
+    try {
+      // Make the request
+      const resp = await searchSource.fetch();
+
+      this.searchScope.isLoading = false;
+
+      // Log response to inspector
+      inspectorRequest.stats(getResponseInspectorStats(searchSource, resp)).ok({ json: resp });
+
+      // Apply the changes to the angular scope
+      this.searchScope.$apply(() => {
+        this.searchScope!.hits = resp.hits.hits;
+        this.searchScope!.totalHitCount = resp.hits.total;
+      });
+    } catch (error) {
+      toastNotifications.addError(error, {
+        title: i18n.translate('kbn.embeddable.errorTitle', {
+          defaultMessage: 'Error fetching data',
+        }),
+      });
+    }
+  };
+
   private pushContainerStateParamsToScope(searchScope: SearchScope) {
+    const isFetchRequired =
```

Comment:
Seems like something is off here, a saved search still fetches new data on a dashboard if a disabled filter is changed